### ISSUE TITLE
Feature Request: VM with MSI authentication enabled

### ISSUE THREAD

**Issue by sebastiankasprzak** [test-kitchen/kitchen-azurerm#93]
## Overview

For some workflows it would be useful if as part of the kitchen converge action, when VM is created in Azure, that MSI gets enabled and configured at the Azure platform level.

### Example config:
```
driver:
  name: azurerm
  subscription_id: '4801fa9d-YOUR-GUID-HERE-b265ff49ce21'
  location: 'West Europe'
  machine_size: 'Standard_D1'
  enable_msi: true
  msi_rbac_scopes: 
    Reader:
      - /subscriptions/4801fa9d-YOUR-GUID-HERE-b265ff49ce21
    Contributor:
      - /subscriptions/4801fa9d-YOUR-GUID-HERE-b265ff49ce21/
      - /subscriptions/4801fa9d-YOUR-GUID-HERE-b265ff49ce21/resourceGroups/myGroup
      - /subscriptions/4801fa9d-YOUR-GUID-HERE-b265ff49ce21/resourceGroups/my2ndGroup/providers/Microsoft.Compute/virtualMachines/myVM
```

This will enable testing of applications utilising MSI authentication and reduce the reliance on hardcoded credentials within cookbooks, attributes and .kitchen.yml files.

I believe that it is not possible to implement this functionality by using `pre_deployment_template` as the target VM will not exist at the time, and MSI auth needs to be enabled as part of VM creation or post VM creation.

**Comment by jcoeltjen** [comment]
Would a generic 'post-deployment-template' parameter also work for you?

**Comment by sebastiankasprzak** [comment]
Yeah that will do the job as well @jcoeltjen  is this the preferred approach?

**Comment by sebastiankasprzak** [comment]
I will have a go at implementing this feature, hopefully PR will be inbound shortly

**Comment by stuartpreston** [comment]
Released in v0.14.4
